content/markdown-viewer.js

```diff
@@ -27,7 +29,62 @@ extensions.markdown = {};
         // we'll recalculate it after the panel is shown.
         return x - (panel.boxObject.width || 120) - 2;
     }
+    
+    /**
+     * Exports DOM document with styles to HTML text
+     * @param {Document} document
+     * @param {string} name (document title)
+     */
+    function exportDocument(document, name) {
+        var s = document.styleSheets,
+            sa, si, ra, ri, rules = [], EOL = "\r\n";
+        for (var si = 0; si < s.length; si++)
+            if (s[si])
+                for (var ri = 0; ri < s[si].cssRules.length; ri++)
+                    rules.push(s[si].cssRules[ri].cssText);
```

Comment:
Nitpick - you are using a lot of <=2 character variable names here, this makes for code that is hard to read over and it really doesn't gain you any measurable performance.